### ISSUE TITLE
Stopping GCP instances takes too long time

### ISSUE THREAD

**Issue by fabioDMFerreira** [nanovms/nanos#1318]
I think we need to handle the stop signal gracefully. After stopping an instance it seems the operation gets stuck. Probably related with #1065.

**Comment by francescolavra** [comment]
@fabioDMFerreira yes, we need to catch and handle the ACPI poweroff signal that is sent by the host to the VM when a GCP instance is stopped. We are tracking this issue in #1065.

**Comment by fabioDMFerreira** [comment]
thanks @francescolavra 

I'll wait for this to solve https://github.com/nanovms/ops/issues/684.

**Comment by fabioDMFerreira** [comment]
@francescolavra there's a regression. Stopping instances in GCP is taking too long again.

Wondering if the reboot on exit feature has some influence.

**Comment by francescolavra** [comment]
@fabioDMFerreira yes, the reboot on exit feature influences the behavior of an instance being stopped.
When this feature is not enabled, stopping an instance takes the usual time (I just measured 20 seconds for `instance stop` and 22 seconds for `instance delete`, in line with the time required to stop or delete a Linux instance via the GCP portal); when this feature is enabled, trying to stop an instance will actually make it reboot, which means that it can only be stopped forcefully by GCP (which it does after a 90 second timeout).
